enterprise/frontend/src/metabase-enterprise/moderation/components/ModerationIssueActionMenu.jsx

```diff
@@ -14,45 +15,74 @@ import { getIsModerator } from "metabase-enterprise/moderation/selectors";
 ModerationIssueActionMenu.propTypes = {
   className: PropTypes.string,
   triggerClassName: PropTypes.string,
-  isModerator: PropTypes.bool.isRequired,
   onAction: PropTypes.func.isRequired,
-  request: PropTypes.object,
+  targetIssueType: PropTypes.string,
+  userType: PropTypes.string.isRequired,
 };
 
 function ModerationIssueActionMenu({
   className,
   triggerClassName,
   onAction,
-  request,
-  isModerator,
+  targetIssueType,
+  userType,
 }) {
-  const userType = getUserTypeTextKey(isModerator);
-  const issueTypes = getModerationIssueActionTypes(isModerator, request);
+  const menuItems = useMemo(() => {
+    return buildActionMenuItems(userType, targetIssueType, onAction);
+  }, [userType, targetIssueType, onAction]);
+
+  const triggerProps = useMemo(() => {
+    return {
+      iconRight: "chevrondown",
+      className: triggerClassName,
+    };
+  }, [triggerClassName]);
 
   return (
     <EntityMenu
       triggerChildren={MODERATION_TEXT[userType].action}
-      triggerProps={{
-        iconRight: "chevrondown",
-        className: triggerClassName,
-      }}
+      triggerProps={triggerProps}
       className={className}
-      items={issueTypes.map(issueType => {
-        const { icon, color } = getModerationStatusIcon(issueType);
-        return {
-          icon,
-          iconSize: 18,
-          className: `text-${color} text-${color}-hover`,
-          action: () => onAction(issueType),
-          title: MODERATION_TEXT[issueType].action,
-        };
-      })}
+      items={menuItems}
     />
   );
 }
 
-const mapStateToProps = (state, props) => ({
-  isModerator: getIsModerator(state, props),
-});
+const mapStateToProps = (state, props) => {
+  return {
+    userType: getUserTypeTextKey(getIsModerator(state, props)),
+  };
+};
```

Comment:
nip: what do you think about keeping `mapStateToProps` above the component? In the same way, we do with prop types. Can have the same benefit: it'd be clear which props are coming from redux and what should be actually passed as a prop

Reply:
true -- moved it to the top